feat(forms): expose form status values to public API

ChrisDahmCS opened the pull request:
Change the public API to include the form status values allowing status comparisons to the existing status constants instead of redefining within applications.

Closes #27389

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [] Tests for the changes have been added (for bug fixes / features) - adding to public_api, are there tests for this?
- [x] Docs have been added / updated (for bug fixes / features) - ~adding to public_api, are there docs for this?~ added @publicApi to the const docs


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
The form status constants are not exposed to the public API. This causes remaking constants/string when comparing form statuses. 

Issue Number: #27389 


## What is the new behavior?

The public API exposes these values from the model, allowing them to be imported/used.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

kara commented on packages/forms/src/model.ts:
```diff
@@ -16,13 +16,17 @@ import {toObservable} from './validators';
+ *
+ * @publicApi
  */
 export const VALID = 'VALID';
```
I think we should avoid exporting something as generic as `VALID` or `DISABLED`. Can we instead add static fields to `AbstractControl` for these? Like `AbstractControl.VALID`?

ChrisDahmCS replied:
Yep, on it (and the tests)!

kara reviewed (changes requested):
We also will need some tests.

kara commented:
@ChrisDahmCS Re public API guard, you'll need to run the bazel test

```
yarn bazel test tools/public_api_guard/...
```

ChrisDahmCS commented:
@kara 
I apologize but I am still a bit unfamiliar with this part of the process. If there is a change that causes the Bazel test to fail, should I be accepting/committing the new "golden file" with the PR as well?

kara commented:
Yes, if the change in the golden file is expected. If we are adding a new public API (like in this case), we would expect that to change.

JoostK commented:
Why not a const enum instead, that tree shakes much better.

Commit:
feat(forms): add static form status fields

Add static form status fields to the abstract control. Now comparisons to the status can use these static fields.

Closes #27389

ChrisDahmCS commented:
So I feel I should stop to ask more questions:
As of right now I need to change the static names back to uppercase if those will be kept. From the comment by @JoostK as well as others on issues like #27389, would it be preferred for me to add or go the route of adding a type/enum?

kara commented:
@ChrisDahmCS I'm good with a const enum too. @IgorMinar Do you have an opinion?

AndrewKushnir commented:
Hi @ChrisDahmCS, thanks for creating this PR. This change seems to be related (and actually blocked by) the work on the strict form types. Strict form types improvements are on the Roadmap, but there is no ETA at this moment. I'm going to close this PR for now and we'll get back to the possibility of exposing statuses as public APIs once strict form type improvements are implemented. Thank you.
